internal/broker/amqpextra/publish.go

```diff
@@ -19,7 +20,7 @@ func (w *Worker) Publish(ctx context.Context, message broker.Publishable) error
 
 	err := w.worker.Publish(ctx, amqp.PublishDto{
 		Exchange:      w.config.Exchange,
-		RoutingKey:    w.config.RoutingKey,
+		RoutingKey:    strings.Join([]string{w.config.RoutingKey, message.Type()}, "."),
```

Comment:
One couæd argue this is a complex way for formatting a string 🙈

```suggestion
		RoutingKey:    fmt.Sprintf("%s.%s", w.config.RoutingKey, message.Type()),
```